core/sql/optimizer/ScanOptimizer.cpp

```diff
@@ -8703,11 +8713,14 @@ void MDAMCostWA::computeDisjunct()
   if(CURRSTMT_OPTDEFAULTS->indexEliminationLevel() != OptDefaults::MINIMUM
      && (!mdamForced_)
 	 && (CmpCommon::getDefault(RANGESPEC_TRANSFORMATION) == DF_ON )
+	 && (CmpCommon::getDefault(MDAM_APPLY_RESTRICTION_CHECK) == DF_ON )
```

Comment:
Yes, the costing is much better and we should trust it more :-).

On Fri, Oct 23, 2015 at 12:50 PM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/optimizer/ScanOptimizer.cpp
> <https://github.com/apache/incubator-trafodion/pull/144#discussion_r42893895>
> :
>
> > @@ -8703,11 +8713,14 @@ void MDAMCostWA::computeDisjunct()
> >    if(CURRSTMT_OPTDEFAULTS->indexEliminationLevel() != OptDefaults::MINIMUM
> >       && (!mdamForced_)
> >  	 && (CmpCommon::getDefault(RANGESPEC_TRANSFORMATION) == DF_ON )
> > +	 && (CmpCommon::getDefault(MDAM_APPLY_RESTRICTION_CHECK) == DF_ON )
>
> The effect of this change is that we will pick MDAM more often? I take it
> the method "checkMDAMadditionalRestriction" is where the problem was?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/144/files#r42893895>.
>



-- 
Regards, --Qifan